react_redux/README.md

```diff
@@ -0,0 +1,540 @@
+    <Foo
+      ref='myRef'
+    />
+
+    // good
+    <Foo
+      ref={(ref) => { this.myRef = ref; }}
+    />
+    ```
+
+## Parentheses
+
+  - Wrap JSX tags in parentheses when they span more than one line. 
+
+    ```typescript
+    // bad
+    render(): JSX.Element {
+      return <MyComponent className='long body' foo='bar'>
+               <MyChild />
+             </MyComponent>;
+    }
+
+    // good
+    render(): JSX.Element {
+      return (
+        <MyComponent className='long body' foo='bar'>
+          <MyChild />
+        </MyComponent>
+      );
+    }
+
+    // good, when single line
+    render(): JSX.Element {
+      const body = <div>hello</div>;
+      return <MyComponent>{body}</MyComponent>;
+    }
+    ```
+
+## Tags
+
+  - Always self-close tags that have no children. 
+
+    ```typescript
+    // bad
+    <Foo className='stuff'></Foo>
+
+    // good
+    <Foo className='stuff' />
+    ```
+
+  - If your component has multi-line properties, close its tag on a new line. 
+
+    ```typescript
+    // bad
+    <Foo
+      bar='bar'
+      baz='baz' />
+
+    // good
+    <Foo
+      bar='bar'
+      baz='baz'
+    />
+    ```
+
+## Methods
+
+  - Use arrow functions to close over local variables.
+
+    ```typescript
+    const ItemList = (props: Props): JSX.Element => {
+      return (
+        <ul>
+          {props.items.map((item, index) => (
+            <Item
+              key={item.key}
+              onClick={() => doSomethingWith(item.name, index)}
+            />
+          ))}
+        </ul>
+      );
+    }
+    ```
+
+  - Bind event handlers for the render method in the constructor. 
+
+    > Why? A bind call in the render path creates a brand new function on every single render.
+
+    ```typescript
+    // bad
+    class extends React.Component<Props, void> {
+      onClickDiv() {
+        // do stuff
+      }
+
+      render(): JSX.Element {
+        return <div onClick={this.onClickDiv.bind(this)} />;
+      }
+    }
+
+    // good
+    class extends React.Component<Props, void> {
+      constructor(props: Props) {
+        super(props);
+
+        this.onClickDiv = this.onClickDiv.bind(this);
+      }
+
+      onClickDiv() {
+        // do stuff
+      }
+
+      render(): JSX.Element {
+        return <div onClick={this.onClickDiv} />;
+      }
+    }
+    ```
+
+  - Do not use underscore prefix for internal methods of a React component.
+    > Why? Underscore prefixes are sometimes used as a convention in other languages to denote privacy. But, unlike those languages, there is no native support for privacy in JavaScript, everything is public. Regardless of your intentions, adding underscore prefixes to your properties does not actually make them private, and any property (underscore-prefixed or not) should be treated as being public. See issues [#1024](https://github.com/airbnb/javascript/issues/1024), and [#490](https://github.com/airbnb/javascript/issues/490) for a more in-depth discussion.
```

Comment:
Here you can add a word about `private` method in typescript?

Reply:
done